modules/tinymce/src/core/main/ts/api/fmt/Format.ts

```diff
@@ -55,15 +58,17 @@ export interface SelectorFormat extends CommonApplyFormat<SelectorFormat> {
 }
 
 export interface CommonRemoveFormat<T> extends CommonFormat<T> {
-  remove?: 'none' | 'empty' | 'all';
+  remove?: 'none' | 'empty' | 'all' | 'convert';
```

Comment:
I don't like the idea of this to be honest. The way I envisioned this would be solved, is to remove the current format rules and then check to see if it matched any other format. If it didn't then it should also remove the node.

Reply:
This also potentially leads to spans without any styles, etc... being left around which isn't desirable.

Reply:
Looking more, I can see why you went with the convert approach, though I still don't think it's a good option. `convert` will specifically only work for inline elements with styles such as this, so it's conflicting with the other types that work for everything (eg if the element had a format that used attributes instead this would still cause issues).

Reply:
I'll take a better look over the weekend and see what other suggestions I can come up with.

Reply:
Thanks for having a look. I am more than happy to take a different approach. I think there are going to have to be some trade offs since it will be nearly impossible to cater to all of the different format possibilities. I'll see if I can think of a better solution that doesn't require adding new `remove` option.

Reply:
To clarify I'm not against adding a new remove format, it's just if we do it needs to work for everything, not just styles+inline elements.